### PR TITLE
Docs: adding release process to README.md

### PR DESCRIPTION
**Pull request by kantai**
This adds a description of the release process to README.md for non-consensus-breaking releases.

Tagging @gregorycoppola, @jcnelson, @pavitthrap, @lgalabru, @reedrosenbluth, @CharlieC3, @wileyj for input -- if you'd like to make changes, please feel free to push commits to this branch!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -324,6 +325,46 @@ You can automatically reformat your commit via:
...
+
+1. A release manager enumerates the PRs or issues that would _block_
+   the release. A label should be applied to each such issue/PR as
+   `2.1.x.y.z-blocker`. The release manager should ping these
+   issue/PR owners for updates on whether or not those issues/PRs have
+   any blockers or are waiting on feedback.
+1. Once all `2.1.x.y.z-blocker` PRs have been merged to `develop`, the release
+   manager will open a `develop -> master` PR.
+1. Immediately following this, the release manager will then create a new tag
+   by manually triggering the [`stacks-blockchain` Github Actions workflow](https://github.com/blockstack/stacks-blockchain/actions/workflows/stacks-blockchain.yml)
```
Comment (gregorycoppola) [inline README.md:340]: is this by clicking "Run Workflow"? Maybe you can add some text or even optionally a picture.

File: README.md [hunk 2]
```diff
@@ -324,6 +325,46 @@ You can automatically reformat your commit via:
...
+   additional release candidates should be tagged. The release manager is responsible
+   for updating the `develop -> master` PR with information about the discovered issues,
+   even if other community members and developers may be addressing the discovered
+   issues.
+
+1. Once the final release candidate has rolled out successfully without issue on the
+   above staging infrastructure, the release manager tags 2 additional `stacks-blockchain`
+   team members to review the `develop -> master` PR.
+
+1. Once reviewed and approved, the release manager merges the PR, and tags the release
```
Comment (gregorycoppola) [inline README.md:363]: How is this different from the first version of the use of this action (line 340)?

File: README.md [hunk 3]
```diff
@@ -71,15 +71,15 @@ The SIPs are now located in the [stacksgov/sips](https://github.com/stacksgov/si
 
 - [x] **Helium** is a developer local setup, mono-node, assembling SIP 001, SIP 002, SIP 004 and SIP 005. With this version, developers can not only run Stacks 2.0 on their development machines, but also write, execute, and test smart contracts. See the instructions below for more details.
 
-- [X] **Neon** is the first version of our public testnet, which shipped in Q2 2020. This testnet added SIP 003, and will be an open-membership public network, where participants will be able to validate and participate in mining testnet blocks.
+- [x] **Neon** is the first version of our public testnet, which shipped in Q2 2020. This testnet added SIP 003, and will be an open-membership public network, where participants will be able to validate and participate in mining testnet blocks.
```
[the remark this answers is not in the document]
Comment (CharlieC3) [reply README.md:74]: Old docs like this definitely needs to be scrubbed, but for the sake of getting this merged sooner it should probably be part of another PR. Maybe @pgray-hiro is able to help here?

These changes in particular were just from [Prettier](https://prettier.io/) in my IDE.

[the remark this answers is not in the document]
Comment (pgray-hiro) [reply README.md:74]: Agree this change is out of scope for this PR - I can help update these in a separate one.

File: README.md (hunk 2 above)
Comment (kantai) [reply README.md:378]: Added more specificity about the -rc versus non-rc tagging in 2279149

File: README.md [hunk 4]
```diff
@@ -324,6 +325,46 @@ You can automatically reformat your commit via:
...
+   manager will open a `develop -> master` PR.
+1. Immediately following this, the release manager will then create a new tag
+   by manually triggering the [`stacks-blockchain` Github Actions workflow](https://github.com/blockstack/stacks-blockchain/actions/workflows/stacks-blockchain.yml)
+   against the `develop` branch, inputting the release candidate tag, `2.1.x.y.z-rc0`,
+   in the Action's input textbox.
+1. Once the release candidate has been built, and docker images, etc. are available,
+   the release manager will notify various ecosystem participants to test the release
+   candidate on various staging infrastructure:
+
+   1. Stacks Foundation staging environments.
```
[the remark this answers is not in the document]
Comment (kantai) [reply README.md:350]: Added a bit more information in 2279149 -- announce the rc in discord, coordinate with devops / other ecosystem participants via Discord DM or e-mail.

File: README.md (hunk 1 above)
Comment (kantai) [reply README.md:342]: Added more specificity in 2279149

File: README.md [hunk 5]
```diff
@@ -324,6 +325,46 @@ You can automatically reformat your commit via:
 $ cargo fmt --all
 ```
 
+## Non-Consensus Breaking Release Process
```
[the remark this answers is not in the document]
Comment (kantai) [reply README.md:328]: Added in 2279149

File: README.md [hunk 6]
```diff
@@ -324,6 +325,63 @@ You can automatically reformat your commit via:
...
+1. If bugs or issues emerge from the rollout on staging infrastructure, the release
+   will be delayed until those regressions are resolved. As regressions are resolved,
+   additional release candidates should be tagged. The release manager is responsible
+   for updating the `develop -> master` PR with information about the discovered issues,
+   even if other community members and developers may be addressing the discovered
+   issues.
+
+1. Once the final release candidate has rolled out successfully without issue on the
+   above staging infrastructure, the release manager tags 2 additional `stacks-blockchain`
+   team members to review the `develop -> master` PR.
```
Comment (lgalabru) [inline README.md:376]: From experience, I find useful to draft that PR early on (one of the very first steps). It helps the release manager getting a good understanding of the deltas for the incoming release, and target testing on specific areas. WDYT?

File: README.md [hunk 7]
```diff
@@ -324,6 +325,63 @@ You can automatically reformat your commit via:
 $ cargo fmt --all
 ```
 
+## Non-Consensus Breaking Release Process
+
+For non-consensus breaking releases, this project uses the following release process:
+
+1. A release manager enumerates the PRs or issues that would _block_
+   the release. A label should be applied to each such issue/PR as
+   `2.1.x.y.z-blocker`. The release manager should ping these
```
Comment (lgalabru) [inline README.md:334]:
```suggestion
   `2.0.x.y.z-blocker`. The release manager should ping these
```

File: README.md [hunk 8]
```diff
@@ -324,6 +325,63 @@ You can automatically reformat your commit via:
...
+   issue/PR owners for updates on whether or not those issues/PRs have
+   any blockers or are waiting on feedback.
+
+1. Once all `2.1.x.y.z-blocker` PRs have been merged to `develop`, the release
```
Comment (lgalabru) [inline README.md:338]:
```suggestion
1. Once all `2.0.x.y.z-blocker` PRs have been merged to `develop`, the release
```

File: README.md [hunk 9]
```diff
@@ -324,6 +325,63 @@ You can automatically reformat your commit via:
...
+   manager will open a `develop -> master` PR.
+
+1. Immediately following this, the release manager will then create a new tag
+   by manually triggering the [`stacks-blockchain` Github Actions workflow](https://github.com/blockstack/stacks-blockchain/actions/workflows/stacks-blockchain.yml)
+   against the `develop` branch, inputting the release candidate tag, `2.1.x.y.z-rc0`,
```
Comment (lgalabru) [inline README.md:343]:
```suggestion
   against the `develop` branch, inputting the release candidate tag, `2.0.x.y.z-rc0`,
```

File: README.md (hunk 6 above)
Comment (kantai) [reply README.md:376]: Yeah, I think that's a good idea, will add.

**Review by gregorycoppola — APPROVED** [review]
LGTM. Thanks for doing this.

I put a note about saying how to pick the version number but can be in a follow up, or in this one.

**Review by jcnelson — APPROVED** [review]
LGTM!

**Review by CharlieC3 — APPROVED** [review]
:shipit:

**Pull request merged**